pkg/app/mutate_app_psp_removal.go

```diff
@@ -55,19 +57,43 @@ func (m *Mutator) mutateConfigForPSPRemoval(ctx context.Context, app v1alpha1.Ap
 		return result, nil
 	}
 
+	if app.Labels[label.AppOperatorVersion] == "0.0.0" || app.Spec.Catalog == "control-plane-catalog" {
```

Comment:
Haha we are totally going to find some WC apps coming from that catalog now

Reply:
I don't think so. Unless something is **very** wrong.

Reply:
I will revert to my earlier thought about checking Namespace though.

Reply:
Sorry that wasn't intended to be a real concern

Reply:
I know, but it got me thinking and I found a case it did not cover. App Bundles in WC namespaces. They do use `app-operator: 0.0.0`.
All collection apps on MCs (including observability bundles) get deployed to `giantswarm` namespace. So that's how we'll differentiate.

Observability bundle is in `default` catalog, so catalog names would not save us.

Reply:
In short: App platform is tricky and has a lot of overrides to default behavior built in for stuff like App bundles. Which is unfortunate and a bitch to test :sweat: